vendor/bat-native-ads/src/bat/ads/internal/container_util.h

```diff
@@ -77,6 +78,23 @@ bool CompareAsSets(const T& c1, const T& c2) {
   return true;
 }
 
+template <typename T>
+std::vector<T> SetIntersection(const std::vector<T>& lhs,
+                               const std::vector<T>& rhs) {
```

Comment:
could we pass vector<T> by value here?
`
std::vector<T> SetIntersection(std::vector<T> lhs,
                                                 std::vector<T> rhs)
`
This would be the same as making mutable vectors.
The benefit is that in some cases we could move vector as argument:

`SetIntersection(std::move(v1), std::move(v2));`